src/sentry/event_manager.py

```diff
@@ -398,27 +416,32 @@ def save(self, project, raw=False):
         from sentry.tasks.post_process import index_event_tags
 
         project = Project.objects.get_from_cache(id=project)
+        assert project is not None, 'did not get a project. this is bad'
 
         data = self.data.copy()
 
         # First we pull out our top-level (non-data attr) kwargs
-        event_id = data.pop('event_id')
-        level = data.pop('level')
-
-        culprit = data.pop('culprit', None)
-        logger_name = data.pop('logger', None)
-        server_name = data.pop('server_name', None)
-        site = data.pop('site', None)
-        checksum = data.pop('checksum', None)
-        fingerprint = data.pop('fingerprint', None)
-        platform = data.pop('platform', None)
-        release = data.pop('release', None)
-        environment = data.pop('environment', None)
-
-        # unused
-        time_spent = data.pop('time_spent', None)
-        message = data.pop('message', '')
+        event_id = data.get('event_id')
+        level = data.get('level')
+
+        culprit = data.get('culprit', None)
+        logger_name = data.get('logger', None)
+        server_name = data.get('server_name', None)
+        site = data.get('site', None)
+        checksum = data.get('checksum', None)
+        fingerprint = data.get('fingerprint', None)
+        platform = data.get('platform', None)
+        release = data.get('release', None)
+        environment = data.get('environment', None)
+        message = data.get('message', '')
+        time_spent = data.get('time_spent', None)
+
+        date = datetime.fromtimestamp(data.get('timestamp'))
+        date = date.replace(tzinfo=timezone.utc)
 
+        # Some processing for when things went well
+        for key in NON_STORED_DATA_KEYS:
+            data.pop(key, None)
```

Comment:
I can back those changes out now that the rest of the reprocessing is changing anyways. It was intended to make it cleaner to work with data because these keys are not initially removed from it.